poetry.lock

```diff
@@ -1162,12 +1174,24 @@ files = [
     {file = "idna-3.4.tar.gz", hash = "sha256:814f528e8dead7d329833b91c5faa87d60bf71824cd12a7530b5526063d02cb4"},
 ]
 
+[[package]]
+name = "imagesize"
+version = "1.4.1"
+description = "Getting image size from png/jpeg/jpeg2000/gif file"
+category = "dev"
+optional = false
+python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*"
+files = [
+    {file = "imagesize-1.4.1-py2.py3-none-any.whl", hash = "sha256:0d8d18d08f840c19d0ee7ca1fd82490fdc3729b7ac93f49870406ddde8ef8d8b"},
+    {file = "imagesize-1.4.1.tar.gz", hash = "sha256:69150444affb9cb0d5cc5a92b3676f0b2fb7cd9ae39e947a5e11a36b4497cd4a"},
+]
+
 [[package]]
 name = "importlib-metadata"
 version = "5.0.0"
 description = "Read metadata from Python packages"
 category = "main"
-optional = true
+optional = false
```

Comment:
There is multiple `optional = False` changes across this file, I think this is an error (poetry is handling group wrong ?)

For instance coming from master, if I checkout on this branch and start a regular install, I should not be installing any new packages, but this is not what happen:
```
$ POETRY_LIBPARSEC_BUILD_STRATEGY=no_build poetry install -E backend -E core
Installing dependencies from lock file

Package operations: 14 installs, 1 update, 0 removals

  • Installing alabaster (0.7.13)
  • Updating docutils (0.17.1 -> 0.18.1)
  • Installing imagesize (1.4.1)
  • Installing pygments (2.14.0)
  • Installing snowballstemmer (2.2.0)
  • Installing sphinxcontrib-applehelp (1.0.4)
  • Installing sphinxcontrib-devhelp (1.0.2)
  • Installing sphinxcontrib-htmlhelp (2.0.1)
  • Installing sphinxcontrib-jsmath (1.0.1)
  • Installing sphinxcontrib-qthelp (1.0.3)
  • Installing sphinxcontrib-serializinghtml (1.1.5)
[...]
```

Reply:
This is most likely because this group is not declared optional, see https://python-poetry.org/docs/master/managing-dependencies/#optional-groups

```ini
[tool.poetry.group.docs]
optional = true

[tool.poetry.group.docs.dependencies]
mkdocs = "*"
```

Reply:
:open_mouth: I will look into that

Reply:
I've revert the change and regenerate the `poetry.lock` with

```shell
poetry lock --no-update
```

It result in the same lock file even with 

```toml
tool.poetry.group.docs.optional = true
```

Reply:
- `charset-normalizer` is required by `requests`
- `importlib-metadata` is required by `sphinx`
- `jinja2` required by `quart` (optional) & `sphinx`

Reply:
I've tried `poetry install -E core` and I don't have `jinja2` installed so it seems to be ok ?

Reply:
So it seems pip install works as expected (i.e. it never installs sphinx&co when `docs` group is not specified):
```shell
$ POETRY_LIBPARSEC_BUILD_STRATEGY=no_build pip install -e .[core,backend]
$ pip freeze > pip_install_new.log
$ diff pip_install_old.log pip_install_new.log  # pip_install_old.log is generated from master
0a1
66c67
< -e git+ssh://git@github.com/Scille/parsec-cloud.git@b1c8817b6d1258b165d4300d9646a9ead9e88d33#egg=UNKNOWN
---
> -e git+ssh://git@github.com/Scille/parsec-cloud.git@7407dd006d45ddf7244b79c769ff45634a079ce7#egg=UNKNOWN
```
```shell
$ POETRY_LIBPARSEC_BUILD_STRATEGY=no_build poetry install -E backend -E core
$ pip freeze > poetry_install_new.log
$ diff poetry_install_old.log poetry_install_new.log  # poetry_install_old.log is generated from master
0a1
25c26
< docutils==0.17.1
---
> docutils==0.18.1
```